File: src/main/kotlin/com/freecourses/service/CoursesService.java
package com.freecourses.service;

import com.freecourses.model.CourseDifficulty;
import com.freecourses.model.ListCoursesResponse;

import javax.validation.constraints.NotNull;
import java.nio.ByteBuffer;

public interface CoursesService {
    ListCoursesResponse getCourses(@NotNull(message = "category can not be null") String category,
                                   String subcategory, CourseDifficulty difficulty, ByteBuffer lastPageToken);
}

File: src/main/kotlin/com/freecourses/service/CoursesService.kt
package com.freecourses.service

import com.freecourses.model.Course
import com.freecourses.model.CourseDifficulty
import com.freecourses.model.CreateCourseRequest
import com.freecourses.model.ListCoursesResponse
import java.util.*

interface CoursesService {
    fun getCourses(
        category: String, subcategory: String?, difficulty: CourseDifficulty?, pageSize: Int, nextPageToken: ByteArray?
    ): ListCoursesResponse

    fun getCourse(courseId: UUID): Course

    fun createCourse(course: CreateCourseRequest): Course
}